### DISCUSSION TITLE
Kotlin issues

### DISCUSSION THREAD

**Discussion by renanboni** [CoderLine/alphaTab#1146] [Q&A] [answered]
Hey, I'm not sure if I should open this since the kotlin release is not stable yet, but just so you know, the app crashes when I open gp4 files (not all of them), with the following error:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: net.alphatab.android, PID: 30721
    java.lang.ClassCastException: No enum with value 9.0 found
        at alphaTab.model.DynamicValue$Companion.fromValue(DynamicValue.kt:83)
```

I'm running the sample you've provided in your repository.

Also, it crashes sometimes randomly with media player errors.

Let me know if you are already aware of the mentioned issues otherwise, I can provide more details.

==========

Another issue that I've faced is that the app crashes if you pause and then play the song:

[stacktrace.txt](https://github.com/CoderLine/alphaTab/files/11203161/stacktrace.txt)

and also if you jump around while the song is playing

```
E/AndroidRuntime: FATAL EXCEPTION: alphaSynthWorkerThread
    Process: net.alphatab.android, PID: 22335
    java.lang.StackOverflowError: stack size 1039KB
        at java.util.ArrayList.ensureCapacityInternal(ArrayList.java:241)
        at java.util.ArrayList.add(ArrayList.java:485)
        at java.util.ArrayList$SubList.add(ArrayList.java:1066)
        at java.util.ArrayList$SubList.add(ArrayList.java:1066)
        at java.util.ArrayList$SubList.add(ArrayList.java:1066)
        at java.util.ArrayList$SubList.add(ArrayList.java:1066)
```

Let me know if you can reproduce those issues otherwise, I can send you a video.

I'm using a Pixel 6

**Comment by Danielku15** [comment]
The error you show doesn't seem very specific to Kotlin but rather a general Guitar Pro 4 thing. Testing old gp4 files and ensuring that everything is correct is always a bit dricky because Arobas made quite some adaptions over the years to the formats. 

Can you provide the full callstack and me the files which cause these issues?

**Reply by renanboni** [reply]
I think it might be related to Kotlin because the same file runs on alphatab web, I attached the gp4 file and later in the day I can post the full callstack

[guns_n_roses-yesterdays.gp4.zip](https://github.com/CoderLine/alphaTab/files/11199257/guns_n_roses-yesterdays.gp4.zip)

**Reply by Danielku15** [reply]
In Kotlin it likely becomes a lot more visible because invalid enumeration values cause an exception. In C# and JavaScript the exception might not happen due to the numeric nature of enums (we can simply cast there) but on all platforms the number 9 is rather an unhandled value on all platforms.

**Reply by renanboni** [reply]
hey @Danielku15 I've updated the ticket with two more scenarios where the app crashes

**Comment by Danielku15** [comment, marked as answer]
Problem is identified. I am working on a fix already. 
https://github.com/CoderLine/alphaTab/issues/1150
